packages/xo-server/src/xo-mixins/vmware/importDisksfromDatastore.mjs

```diff
@@ -29,15 +32,23 @@ const importDiskChain = Disposable.factory(async function* importDiskChain(
         throw new Error(`Can't import delta of a running VM without its parent VHD`)
       }
       vhd = yield openDeltaVmdkasVhd(datastoreName, path + '/' + fileName, parentVhd, {
-        lookMissingBlockInParent: isFullImport, // only look to missing block on full import
+        lookMissingBlockInParent, // only look to missing block on full import
         esxi,
         dataStoreToHandlers,
       })
     }
+    // is this disk already imported ?
+    const previouslyImportedVdi = diskIsAlreadyImported(sr, disk)
```

Comment:
`diskIsAlreadyImported` return an array of VDIs, so I believe you have some JS error.
For example, line `79` you are doing `vdi.$importContent` and line `81` `vdi.$ref`.